middleware/proxy/upstream.go

```diff
@@ -150,6 +153,14 @@ func RegisterPolicy(name string, policy func() Policy) {
 	supportedPolicies[name] = policy
 }
 
+// AddProxyHeader adds a proxy header.
+func addProxyHeader(header, value string) {
+	if proxyHeaders == nil {
+		proxyHeaders = make(map[string][]string)
```

Comment:
Sure. I was just following what was there. I figured the logic was to do lazy instantiation to avoid the overhead, since the reverse proxy might not add any proxy header directives.